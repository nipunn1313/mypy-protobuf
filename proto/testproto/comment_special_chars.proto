syntax = "proto3";

package comment_special_chars;

message Test {
  // Ending with "
  string a = 1;
  // Ending with ""
  string b = 2;
  // Ending with """
  string c = 3;
  // Ending with \
  string d = 4;
  // Containing bad escape: \x
  string e = 5;
  // Containing """" quadruple
  string f = 6;
  // Containing """"" quintuple
  string g = 7;
  // Containing """""" sextuple
  string h = 8;
  // """ Multiple """ triples """
  string i = 9;
  // "quotes" can be a problem in comments.
  // """Triple quotes""" just as well
  string j = 10;
  // """"""""""""""""""""""""""""""""""""""""""""""""
  // "                                              "
  // " Super Duper comments with surrounding edges! "
  // "                                              "
  // "            Pay attention to me!!!!           "
  // "                                              "
  // """"""""""""""""""""""""""""""""""""""""""""""""
  string k = 11;
}
