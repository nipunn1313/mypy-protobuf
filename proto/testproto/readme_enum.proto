syntax = "proto3";

package test;

enum MyEnum {
  HELLO = 0;
  WORLD = 1;
}
