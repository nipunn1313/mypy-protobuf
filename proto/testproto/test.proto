syntax = "proto2";

package test;

import "mypy_protobuf/extensions.proto";
import "testproto/inner/inner.proto";
import "testproto/nested/nested.proto";
import "testproto/nopackage.proto";
import "testproto/test3.proto";

// Outer Enum
enum OuterEnum {
  // FOO
  FOO = 1;
  // BAR
  BAR = 2;
}

// Naming conflicts!
enum NamingConflicts {
  Name = 1;
  Value = 2;
  keys = 3;
  values = 4;
  items = 5;
  // See https://github.com/protocolbuffers/protobuf/issues/8803
  // proto itself generates broken code when DESCRIPTOR is there
  // DESCRIPTOR = 8;
}

// Message with one of everything
message Simple1 {
  // Inner Enum
  enum InnerEnum {
    // INNER1
    INNER1 = 1;
    // INNER2
    INNER2 = 2;
  }
  message InnerMessage {}

  optional string a_string = 1;
  repeated string a_repeated_string = 2;
  optional bool a_boolean = 3;
  optional uint32 a_uint32 = 4;
  optional OuterEnum a_enum = 5;
  optional test3.OuterEnum a_external_enum = 6;

  optional inner.Inner a_inner = 7;
  optional test.nested_package.Nested a_nested = 12;

  optional InnerEnum inner_enum = 8;
  repeated InnerEnum rep_inner_enum = 9;
  optional InnerMessage inner_message = 10;
  repeated InnerMessage rep_inner_message = 11;

  optional NoPackage no_package = 13;
  optional test.nested_package.AnotherNested.NestedEnum nested_enum = 14;
  optional test.nested_package.AnotherNested.NestedMessage nested_message = 15;

  oneof a_oneof {
      string a_oneof_1 = 16;
      string a_oneof_2 = 17;
      Simple2 outer_message_in_oneof = 18;
      OuterEnum outer_enum_in_oneof = 19;
      InnerEnum inner_enum_in_oneof = 20;
  }

  optional uint32 user_id = 21 [(mypy_protobuf.casttype)="test/test_generated_mypy.UserId"];
  optional string email = 22 [(mypy_protobuf.casttype)="test/test_generated_mypy.Email"];
  map<uint32, string> email_by_uid = 23 [
    (mypy_protobuf.keytype)="test/test_generated_mypy.UserId",
    (mypy_protobuf.valuetype)="test/test_generated_mypy.Email"
  ];

  extensions 1000 to max;
}

message Simple2 {
  required string a_string = 1;

  extensions 1000 to max;
}

message Extensions1 {
  extend Simple1 {
    // ext
    optional Extensions1 ext = 1000;
  }

  optional string ext1_string = 1;
}

message Extensions2 {
  extend Simple1 {
    // foo
    optional Extensions2 foo = 2020;
  }

  optional bool flag = 1;
}

message None {
  optional int64 valid = 1;
}

message PythonReservedKeywords {
  enum finally {
      continue = 1;
      valid_in_finally = 2;
  }

  message lambda {
      optional int64 continue = 1;
      optional int64 valid = 2;
  }

  required int64 from = 1;
  optional Simple2 in = 2;
  optional finally is = 3;
  optional int64 for = 5;
  optional int64 try = 6;
  optional int64 def = 7;
  optional int64 nonlocal = 8;
  optional int64 while = 9;
  optional int64 and = 10;
  optional int64 del = 11;
  optional int64 global = 12;
  optional int64 not = 13;
  optional int64 with = 14;
  optional int64 as = 15;
  optional int64 elif = 16;
  optional int64 if = 17;
  optional int64 or = 18;
  optional int64 yield = 19;
  optional int64 assert = 20;
  optional int64 else = 21;
  optional int64 import = 22;
  optional int64 pass = 23;
  optional int64 break = 24;
  optional int64 except = 25;
  optional int64 raise = 26;
  optional int64 False = 27;
  optional int64 True = 29;
  optional int64 class = 30;

  // Test unreserved identifiers w/ reserved message names
  optional None none = 28;
  optional finally valid = 31;
}

// Do one with just one arg - to make sure it's syntactically correct
message PythonReservedKeywordsSmall {
  required int64 from = 1;
}

// Method name is reserved
service PythonReservedKeywordsService {
    // lambda
    rpc lambda(Simple1) returns (PythonReservedKeywords.lambda) {}
    // valid_method_name1
    rpc valid_method_name1(Simple1) returns (None) {}
    // valid_method_name2
    rpc valid_method_name2(Simple1) returns (PythonReservedKeywords.lambda) {}
}

// when service name itself is reserved - generated code was found to be invalid
// in protoc 3.17.3
//service global {
//    rpc Echo(Simple1) returns (Simple2) {}
//}

option py_generic_services = true;
service ATestService {
    rpc Echo(Simple1) returns (Simple2) {}
}

message SelfField {
    // Field self -> must generate an __init__ method w/ different name
    optional int64 self = 1;
}
