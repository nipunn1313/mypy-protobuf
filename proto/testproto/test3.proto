syntax = "proto3";

package test3;

enum OuterEnum {
  UNKNOWN = 0;
  FOO3 = 1;
  BAR3 = 2;
}

message OuterMessage3 {
  string a_string = 1;
}

message SimpleProto3 {
  enum InnerEnum {
    INNER1 = 0;
    INNER2 = 1;
  }

  string a_string = 1;
  repeated string a_repeated_string = 2;
  OuterEnum a_outer_enum = 3;
  OuterMessage3 outer_message = 4;
  InnerEnum inner_enum = 15;

  oneof a_oneof {
      string a_oneof_1 = 5;
      string a_oneof_2 = 6;
      OuterMessage3 outer_message_in_oneof = 12;
      OuterEnum outer_enum_in_oneof = 13;
      InnerEnum inner_enum_in_oneof = 14;
  }

  oneof b_oneof {
      string b_oneof_1 = 7;
      string b_oneof_2 = 8;
  }

  OuterMessage3 bool = 9;

  // Test having fieldname match messagename
  OuterEnum OuterEnum = 10;
  OuterMessage3 OuterMessage3 = 11;

  // Test generation of map
  map<int32, string> map_scalar = 16;
  map<int32, OuterMessage3> map_message = 17;

  optional string an_optional_string = 18;
}
