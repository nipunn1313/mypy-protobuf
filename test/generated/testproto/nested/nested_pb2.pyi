"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import testproto.test3_pb2
import typing
import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor

class Nested(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor

    A_FIELD_NUMBER: builtins.int
    a: testproto.test3_pb2.OuterEnum.ValueType
    def __init__(
        self,
        *,
        a: testproto.test3_pb2.OuterEnum.ValueType = ...,
    ) -> None: ...
    def ClearField(self, field_name: typing_extensions.Literal["a", b"a"]) -> None: ...

global___Nested = Nested

class AnotherNested(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor

    class _NestedEnum:
        ValueType = typing.NewType("ValueType", builtins.int)
        V: typing_extensions.TypeAlias = ValueType

    class _NestedEnumEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[AnotherNested._NestedEnum.ValueType], builtins.type):  # noqa: F821
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor
        INVALID: AnotherNested._NestedEnum.ValueType  # 0
        ONE: AnotherNested._NestedEnum.ValueType  # 1
        TWO: AnotherNested._NestedEnum.ValueType  # 2

    class NestedEnum(_NestedEnum, metaclass=_NestedEnumEnumTypeWrapper): ...
    INVALID: AnotherNested.NestedEnum.ValueType  # 0
    ONE: AnotherNested.NestedEnum.ValueType  # 1
    TWO: AnotherNested.NestedEnum.ValueType  # 2

    class NestedMessage(google.protobuf.message.Message):
        DESCRIPTOR: google.protobuf.descriptor.Descriptor

        class _NestedEnum2:
            ValueType = typing.NewType("ValueType", builtins.int)
            V: typing_extensions.TypeAlias = ValueType

        class _NestedEnum2EnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[AnotherNested.NestedMessage._NestedEnum2.ValueType], builtins.type):  # noqa: F821
            DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor
            UNDEFINED: AnotherNested.NestedMessage._NestedEnum2.ValueType  # 0
            NESTED_ENUM1: AnotherNested.NestedMessage._NestedEnum2.ValueType  # 1
            NESTED_ENUM2: AnotherNested.NestedMessage._NestedEnum2.ValueType  # 2

        class NestedEnum2(_NestedEnum2, metaclass=_NestedEnum2EnumTypeWrapper): ...
        UNDEFINED: AnotherNested.NestedMessage.NestedEnum2.ValueType  # 0
        NESTED_ENUM1: AnotherNested.NestedMessage.NestedEnum2.ValueType  # 1
        NESTED_ENUM2: AnotherNested.NestedMessage.NestedEnum2.ValueType  # 2

        S_FIELD_NUMBER: builtins.int
        B_FIELD_NUMBER: builtins.int
        NE_FIELD_NUMBER: builtins.int
        NE2_FIELD_NUMBER: builtins.int
        s: builtins.str
        b: builtins.bool
        ne: global___AnotherNested.NestedEnum.ValueType
        ne2: global___AnotherNested.NestedMessage.NestedEnum2.ValueType
        def __init__(
            self,
            *,
            s: builtins.str = ...,
            b: builtins.bool = ...,
            ne: global___AnotherNested.NestedEnum.ValueType = ...,
            ne2: global___AnotherNested.NestedMessage.NestedEnum2.ValueType = ...,
        ) -> None: ...
        def ClearField(self, field_name: typing_extensions.Literal["b", b"b", "ne", b"ne", "ne2", b"ne2", "s", b"s"]) -> None: ...

    def __init__(
        self,
    ) -> None: ...

global___AnotherNested = AnotherNested
