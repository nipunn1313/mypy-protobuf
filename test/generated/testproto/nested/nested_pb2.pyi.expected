"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import testproto.test3_pb2
import typing
import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor = ...

class Nested(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    A_FIELD_NUMBER: builtins.int
    a: testproto.test3_pb2.OuterEnum.V = ...
    def __init__(self,
        *,
        a : testproto.test3_pb2.OuterEnum.V = ...,
        ) -> None: ...
    def ClearField(self, field_name: typing_extensions.Literal[u"a",b"a"]) -> None: ...
global___Nested = Nested

class AnotherNested(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    class NestedEnum(metaclass=_NestedEnum):
        V = typing.NewType('V', builtins.int)

    INVALID = AnotherNested.NestedEnum.V(0)
    ONE = AnotherNested.NestedEnum.V(1)
    TWO = AnotherNested.NestedEnum.V(2)

    class _NestedEnum(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[NestedEnum.V], builtins.type):  # type: ignore
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
        INVALID = AnotherNested.NestedEnum.V(0)
        ONE = AnotherNested.NestedEnum.V(1)
        TWO = AnotherNested.NestedEnum.V(2)

    class NestedMessage(google.protobuf.message.Message):
        DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
        class NestedEnum2(metaclass=_NestedEnum2):
            V = typing.NewType('V', builtins.int)

        UNDEFINED = AnotherNested.NestedMessage.NestedEnum2.V(0)
        NESTED_ENUM1 = AnotherNested.NestedMessage.NestedEnum2.V(1)
        NESTED_ENUM2 = AnotherNested.NestedMessage.NestedEnum2.V(2)

        class _NestedEnum2(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[NestedEnum2.V], builtins.type):  # type: ignore
            DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
            UNDEFINED = AnotherNested.NestedMessage.NestedEnum2.V(0)
            NESTED_ENUM1 = AnotherNested.NestedMessage.NestedEnum2.V(1)
            NESTED_ENUM2 = AnotherNested.NestedMessage.NestedEnum2.V(2)

        S_FIELD_NUMBER: builtins.int
        B_FIELD_NUMBER: builtins.int
        NE_FIELD_NUMBER: builtins.int
        NE2_FIELD_NUMBER: builtins.int
        s: typing.Text = ...
        b: builtins.bool = ...
        ne: global___AnotherNested.NestedEnum.V = ...
        ne2: global___AnotherNested.NestedMessage.NestedEnum2.V = ...
        def __init__(self,
            *,
            s : typing.Text = ...,
            b : builtins.bool = ...,
            ne : global___AnotherNested.NestedEnum.V = ...,
            ne2 : global___AnotherNested.NestedMessage.NestedEnum2.V = ...,
            ) -> None: ...
        def ClearField(self, field_name: typing_extensions.Literal[u"b",b"b",u"ne",b"ne",u"ne2",b"ne2",u"s",b"s"]) -> None: ...

    def __init__(self,
        ) -> None: ...
global___AnotherNested = AnotherNested
